### ISSUE TITLE
4: Custom Logistic Regression Classifier

### ISSUE THREAD

**Issue by jenn-le** [Thakugan/machine-learning-notebooks#26]
[20 points] Create a custom, one-versus-all logistic regression classifier using numpy and scipy to optimize. Use object oriented conventions identical to scikit-learn. You should start with the template developed by the instructor in the course. You should add the following functionality to the logistic regression classifier:
Ability to choose optimization technique when class is instantiated: either steepest descent, stochastic gradient descent, or Newton's method. 
Update the gradient calculation to include a customizable regularization term (either using no regularization, L1 regularization, L2 regularization, or both L1/L2 norm of the weights). Associate a cost with the regularization term, "C", that can be adjusted when the class is instantiated.  

Exceptional Work (10 points total)
You have free reign to provide additional analyses.
One idea (required for 7000 level students): Make your implementation of logistic regression compatible with the GridSearchCV function that is part of scikit-learn.
